application/src/main/java/org/togetherjava/tjbot/routines/ModAuditLogRoutine.java

```diff
@@ -0,0 +1,371 @@
+        // NOTE Temporary bans are realized as permanent bans with automated unban,
+        // hence we can not differentiate a permanent or a temporary ban here
+        return Optional.of(handleAction(Action.BAN, entry));
+    }
+
+    @SuppressWarnings("MethodMayBeStatic")
+    private @NotNull Optional<RestAction<MessageEmbed>> handleUnbanEntry(
+            @NotNull AuditLogEntry entry) {
+        return Optional.of(handleAction(Action.UNBAN, entry));
+    }
+
+    @SuppressWarnings("MethodMayBeStatic")
+    private @NotNull Optional<RestAction<MessageEmbed>> handleKickEntry(
+            @NotNull AuditLogEntry entry) {
+        return Optional.of(handleAction(Action.KICK, entry));
+    }
+
+    private @NotNull Optional<RestAction<MessageEmbed>> handleRoleUpdateEntry(
+            @NotNull AuditLogEntry entry) {
+        if (containsMutedRole(entry, AuditLogKey.MEMBER_ROLES_ADD)) {
+            return handleMuteEntry(entry);
+        }
+        if (containsMutedRole(entry, AuditLogKey.MEMBER_ROLES_REMOVE)) {
+            return handleUnmuteEntry(entry);
+        }
+        return Optional.empty();
+    }
+
+    @SuppressWarnings("MethodMayBeStatic")
+    private @NotNull Optional<RestAction<MessageEmbed>> handleMuteEntry(
+            @NotNull AuditLogEntry entry) {
+        // NOTE Temporary mutes are realized as permanent mutes with automated unmute,
+        // hence we can not differentiate a permanent or a temporary mute here
+        return Optional.of(handleAction(Action.MUTE, entry));
+    }
+
+    @SuppressWarnings("MethodMayBeStatic")
+    private @NotNull Optional<RestAction<MessageEmbed>> handleUnmuteEntry(
+            @NotNull AuditLogEntry entry) {
+        return Optional.of(handleAction(Action.UNMUTE, entry));
+    }
+
+    @SuppressWarnings("MethodMayBeStatic")
+    private @NotNull Optional<RestAction<MessageEmbed>> handleMessageDeleteEntry(
+            @NotNull AuditLogEntry entry) {
+        return Optional.of(handleAction(Action.MESSAGE_DELETION, entry));
+    }
+
+    private boolean containsMutedRole(@NotNull AuditLogEntry entry, @NotNull AuditLogKey key) {
+        List<Map<String, String>> roleChanges = Optional.ofNullable(entry.getChangeByKey(key))
+            .<List<Map<String, String>>>map(AuditLogChange::getNewValue)
+            .orElse(List.of());
+        return roleChanges.stream()
+            .map(Map::entrySet)
+            .flatMap(Collection::stream)
+            .filter(changeEntry -> "name".equals(changeEntry.getKey()))
+            .map(Map.Entry::getValue)
+            .anyMatch(isMutedRole);
+    }
+
+    private Optional<TextChannel> getModAuditLogChannel(@NotNull Guild guild) {
+        // Check cache first, then get full list
+        return guild.getTextChannelCache()
+            .stream()
+            .filter(isAuditLogChannel)
+            .findAny()
+            .or(() -> guild.getTextChannels().stream().filter(isAuditLogChannel).findAny());
+    }
+
+    private enum Action {
```

Comment:
What about `purge` and `prune`? Are they included in `MESSAGE_DELETION`?

Reply:
yes, any sort of message deletion will trigger this

Reply:
prune is not included (but could be). i dont think we want to monitor that though. at least i would not be interested in prunes.